src/NuGetForUnity/Editor/NugetPackageV2.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace NugetForUnity
+{
+    /// <summary>
+    ///     Represents a package available from NuGet.
+    /// </summary>
+    [Serializable]
+    internal sealed class NugetPackageV2 : NugetPackageV2Base
+    {
+        [SerializeField]
+        private NugetPackageSourceV2 packageSourceV2;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="NugetPackageV2" /> class.
+        /// </summary>
+        /// <param name="packageSourceV2">The source this package was downloaded with / provided by.</param>
+        public NugetPackageV2(NugetPackageSourceV2 packageSourceV2)
+        {
+            this.packageSourceV2 = packageSourceV2;
+        }
+
+        /// <inheritdoc />
+        public override INugetPackageSource PackageSource => packageSourceV2;
+
+        /// <inheritdoc />
+        public override List<NugetPackageVersion> Versions => null;
```

Comment:
I just noticed that this returns null but on line 40 we are passing NugetPackageV2 to FillFromNuspec method that has this line: `package.Versions.Add(package.PackageVersion);` which will cause NullReferenceException, right?

Reply:
you are right. It will be fixed in #568